### PR TITLE
fix: support function imports for odata-v4

### PR DESCRIPTION
**Pull request by florian-richter**
Please provide a description of what your change does and why it is needed.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org).
- If applicable:
  - [x] Documented public API (TypeDoc).
  - [x] Checked that `yarn run doc` still works.

### CODE REVIEW

File: test-packages/e2e-tests/srv/test-service.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ module.exports = async srv => {
     oRequest.reply('SapCloudSdk');
   });
 
+  srv.on('returnSapCloudSdk2', async oRequest => {
+    oRequest.reply('SapCloudSdk');
```
Comment (jjtang1985) [inline test-packages/e2e-tests/srv/test-service.js]: minor: This function has two parameters. Maybe we can return some value related to the parameters.

File: packages/core/src/odata-v2/legacy-request-configs/odata-function-import-request-config.ts [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-  ODataFunctionImportRequestConfig as Base
-} from '../../odata-common';
-
-/**
- * @deprecated Since v1.21.0. Use superclass instead.
- * OData function import request configuration for a set of parameters.
- *
- * @typeparam ParametersT - Type of the original parameters object
- */
-class ODataFunctionImportRequestConfig<ParametersT> extends Base<ParametersT> {
```
Comment (jjtang1985) [inline packages/core/src/odata-v2/legacy-request-configs/odata-function-import-request-config.ts:14]: This class is exported with new name, see the last line of this file.
If I remember correctly, we keep them for compatibility reasons.
@marikaner , please correct me.

File: packages/core/src/odata-v2/request/odata-function-import-request-config.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    oDataUri: ODataUri
+  ) {
+    super(method, defaultServicePath, functionImportName, parameters, oDataUri);
+  }
+
+  resourcePath(): string {
+    return this.functionImportName;
+  }
+
+  queryParameters(): Record<string, any> {
```
Comment (jjtang1985) [inline packages/core/src/odata-v2/request/odata-function-import-request-config.ts:35]: I guess for v2, we keep the same as the previous common

File: packages/core/src/odata-v2/request/odata-function-import-request-config.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  ParametersT
+> extends ODataFunctionImportRequestConfig<ParametersT> {
+  /**
+   * Creates an instance of ODataFunctionImportRequestConfig.
+   *
+   * @param method - HTTP method for the request
+   * @param defaultServicePath - Default path of the service
+   * @param functionImportName - The name of the function import.
+   * @param parameters - Object containing the parameters with a value and additional meta information
+   * @param oDataUri - ???
```
Comment (jjtang1985) [inline packages/core/src/odata-v2/request/odata-function-import-request-config.ts:19]: Please fix it by e.g. searching for the doc of `oDataUri` in other files

File: packages/core/src/odata-v4/request/odata-function-import-request-config.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  ParametersT
+> extends ODataFunctionImportRequestConfig<ParametersT> {
+  /**
+   * Creates an instance of ODataFunctionImportRequestConfig.
+   *
+   * @param method - HTTP method for the request
+   * @param defaultServicePath - Default path of the service
+   * @param functionImportName - The name of the function import.
+   * @param parameters - Object containing the parameters with a value and additional meta information
+   * @param oDataUri - ???
```
Comment (jjtang1985) [inline packages/core/src/odata-v4/request/odata-function-import-request-config.ts:19]: also here

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
LGTM, please consider my comments.
Please also update the change log.

**Comment by jjtang1985** [comment]
For the failing unit test, since `ODataFunctionImportRequestConfig` is now abstract class, we might want to switch to `ODataFunctionImportRequestConfigV2`.

File: packages/core/src/odata-v2/request/odata-function-import-request-config.ts (hunk 3 above)
Comment (florian-richter) [reply packages/core/src/odata-v2/request/odata-function-import-request-config.ts:35]: yes, the new v2 is basically a copy of the file that was in common.

File: packages/core/src/odata-v2/request/odata-function-import-request-config.ts (hunk 4 above)
Comment (florian-richter) [reply packages/core/src/odata-v2/request/odata-function-import-request-config.ts:19]: Oops, forgot about that.

File: packages/core/src/odata-v2/legacy-request-configs/odata-function-import-request-config.ts (hunk 2 above)
Comment (florian-richter) [reply packages/core/src/odata-v2/legacy-request-configs/odata-function-import-request-config.ts:14]: I restored the file and changed it to extend the V2 version of the config.

File: packages/core/src/odata-v2/legacy-request-configs/index.ts [hunk 6]
```diff
@@ -1,7 +1,6 @@
 export * from './odata-batch-request-config';
 export * from './odata-create-request-config';
 export * from './odata-delete-request-config';
-export * from './odata-function-import-request-config';
```
Comment (jjtang1985) [inline packages/core/src/odata-v2/legacy-request-configs/index.ts:4]: I guess you want to revert this change as well.

File: CHANGELOG.md [hunk 7]
```diff
@@ -19,7 +19,7 @@
 
 ## New Functionality
 
--
+- [OData] Support for function imports with and without parameters using the OData V4 syntax.
```
Comment (jjtang1985) [inline CHANGELOG.md:22]: Although the whole feature was missing, to be fair, we announced that the v4 function/action was completed (OKR), we might want to move this line to bug fix.

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
Please consider my comments and the PR title should be `fix:*` as this is a bug fix.

File: CHANGELOG.md (hunk 7 above)
Comment (florian-richter) [reply CHANGELOG.md]: Okay, I understand that if this was announced already, then it should be counted as a bug fix. Wasn't aware.

**Review by jjtang1985 — APPROVED** [review]

**Comment by jjtang1985** [comment]
https://github.com/SAP/cloud-sdk-js/issues/682
